packages/flutter_tools/test/commands.shard/hermetic/run_test.dart

```diff
@@ -261,6 +261,44 @@ void main() {
         Cache: () => Cache.test(processManager: FakeProcessManager.any()),
       });
 
+      testUsingContext('Using flutter run -d all with a single MacOSDesignedForIPadDevices throws a tool error', () async {
+        final RunCommand command = RunCommand();
+        testDeviceManager.devices = <Device>[FakeMacDesignedForIpadDevice()];
+
+        await expectLater(
+                () => createTestCommandRunner(command).run(<String>[
+              'run',
+              '-d',
+              'all',
+            ]), throwsToolExit(message: 'Mac Designed for iPad is currently not supported for flutter run -d'));
+      }, overrides: <Type, Generator>{
+        FileSystem: () => fs,
+        ProcessManager: () => FakeProcessManager.any(),
+        DeviceManager: () => testDeviceManager,
+        Stdio: () => FakeStdio(),
+        Cache: () => Cache.test(processManager: FakeProcessManager.any()),
+      });
+
+      testUsingContext('Using flutter run -d all with MacOSDesignedForIPadDevices removes from device list, and attempts to launch', () async {
+        final RunCommand command = TestRunCommandThatOnlyValidates();
+        testDeviceManager.devices = <Device>[FakeMacDesignedForIpadDevice(), FakeDevice()];
+
+        await createTestCommandRunner(command).run(<String>[
+          'run',
+          '-d',
+          'all',
+        ]);
+
+        expect(command.devices?.length, 1);
+        expect(command.devices?.first.id , 'fake_device');
```

Comment:
`single` will throw if there's more than one in the iterable.
```suggestion
        expect(command.devices?.single.id , 'fake_device');
```

Reply:
oh fun!! i had no idea that existed